internals/ingester/aggregate.go

```diff
@@ -0,0 +1,106 @@
+package ingester
+
+import (
+	"errors"
+	"github.com/myrteametrics/myrtea-engine-api/v5/internals/evaluator"
+	"github.com/myrteametrics/myrtea-engine-api/v5/internals/metrics"
+	"github.com/myrteametrics/myrtea-engine-api/v5/internals/scheduler"
+	"github.com/myrteametrics/myrtea-engine-api/v5/internals/tasker"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+)
+
+// AggregateIngester is a component which process scheduler.ExternalAggregate
+type AggregateIngester struct {
+	Data             chan []scheduler.ExternalAggregate
+	metricQueueGauge *stdprometheus.Gauge
+}
+
+var (
+	_aggregateIngesterGauge = _newRegisteredGauge()
+)
+
+func _newRegisteredGauge() *stdprometheus.Gauge {
+	var gauge = stdprometheus.NewGauge(stdprometheus.GaugeOpts{
+		Namespace:   metrics.MetricNamespace,
+		ConstLabels: metrics.MetricPrometheusLabels,
+		Name:        "aggregateingester_queue",
+		Help:        "this is the help string for aggregateingester_queue",
+	})
+
+	// Register metrics
+	stdprometheus.MustRegister(gauge)
+	gauge.Set(0)
+
+	return &gauge
+}
+
+// NewAggregateIngester returns a pointer to a new AggregateIngester instance
+func NewAggregateIngester() *AggregateIngester {
+	return &AggregateIngester{
+		Data:             make(chan []scheduler.ExternalAggregate, viper.GetInt("AGGREGATEINGESTER_QUEUE_BUFFER_SIZE")),
+		metricQueueGauge: _aggregateIngesterGauge,
+	}
+}
+
+// Run is the main routine of a TypeIngester instance
+func (ingester *AggregateIngester) Run() {
+	zap.L().Info("Starting AggregateIngester")
+
+	for ir := range ingester.Data {
+		zap.L().Debug("Received ExternalAggregate", zap.Int("ExternalAggregate items count", len(ir)))
+
+		err := HandleAggregates(ir)
+		if err != nil {
+			zap.L().Error("AggregateIngester: could not handle aggregates", zap.Error(err))
+		}
+
+		// Update queue gauge
+		(*ingester.metricQueueGauge).Set(float64(len(ingester.Data)))
```

Comment:
Possible de faire directement ça ?
```suggestion
		ingester.metricQueueGauge.Set(float64(len(ingester.Data)))
```